engine/src/distribution/primitives.cpp

```diff
@@ -541,8 +537,8 @@ void collectLeftRightTableSizeBytes(Context * context,	std::vector<int64_t> & no
 		assert(num_bytes_data->view().num_columns() == 1);
 		assert(num_bytes_data->view().num_rows() == 2);
 		
-		std::vector<cudf::size_type> host_data(num_bytes_data->view().column(0).size());
-		CUDA_TRY(cudaMemcpy(host_data.data(), num_bytes_data->view().column(0).data<cudf::size_type>(), num_bytes_data->view().column(0).size() * sizeof(cudf::size_type), cudaMemcpyDeviceToHost));
+		std::vector<int64_t> host_data(num_bytes_data->view().column(0).size());
```

Comment:
is it intended not to use size_type here? I am guessing because number of bytes can be higher than size_type?

Reply:
that is correct. I made a change here because previously the collect and the distribute of the number of bytes, once was sending int64_t and one was collecting cudf::size_type and the difference was causing a problem. And yes you can have an overflow for the number of bytes if you only use cudf::size_type